### PR TITLE
refactor: input component improvements #P23-200

### PR DESCRIPTION
**Pull request by wyola**
- improvements discussed on slack
   - accessibility of error messages, 
   - prevent form submitting on icon button click
   - reserve additional space for supporting label so that inputs does not shift when stacked in the form
- slight animation delay was added to eliminate label jumping when the user clicks on the icon button while the password input is empty ⬇️ 

https://user-images.githubusercontent.com/80450315/229596033-6d96daaf-4a6b-4c9e-9882-996c60971f54.mp4

### CODE REVIEW

**Review by bdeluga — APPROVED** [review]
Looks good 🚀

**Review by Alessiass — APPROVED** [review]
Good change!

File: packages/ui/Input/index.tsx [hunk 1]
```diff
@@ -38,10 +42,10 @@ export const Input = ({
     if (type === "password") {
       return (
         <StyledIcon
-          onClick={(e) => {
-            e.preventDefault();
-            setTypeOverride(typeOverride ? "" : "text");
+          onClick={(event) => {
+            event.preventDefault();
```
Comment (Noshi96) [inline packages/ui/Input/index.tsx:46]: When we type something and click on the eye icon, focus sets our cursor to the beginning of the input. When we use stopPropagation() the cursor stays where we left it. TBH I don't know if this is a good approach. E.g. when we click show password on the Netflix form, the cursor behaves as it does in our application.
```suggestion
            event.stopPropagation();
```

**Review by Noshi96 — APPROVED** [review]

**Review by monika-klokosinska-intive — APPROVED** [review]

File: packages/ui/Input/index.tsx (hunk 1 above)
Comment (wyola) [reply packages/ui/Input/index.tsx:46]: `event.preventDefault()` was used to stop form submission when clicking icon button https://intive-apu8934.slack.com/archives/C04R6LM4JSV/p1680170725637589?thread_ts=1680170425.499899&cid=C04R6LM4JSV

from what I'm reading, `stopPropagation()` does not stop default behaviours

**Pull request merged**